### PR TITLE
Hide status in the whole report with less than 3

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

In the current month of Monthly stats (November 2023), the total figures for Deferrals and Offer conditions not met is 1 which means we fear it's a GDPR breach that the perturbations the D&I team have put in place will not work (any doubts about perturbations talk with D&I team).

## Guidance to review

1. Does Deferred and Conditions not met disappear from the report?
2. Does the CSV also hides the same statuses?

### CODE REVIEW

File: app/models/publications/monthly_statistics/statistics_data_processor.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+module Publications
+  module MonthlyStatistics
+    class StatisticsDataProcessor
+      LOWER_STATS_THAT_VIOLATES_GDPR = 3
```
Comment (vassyz) [inline app/models/publications/monthly_statistics/statistics_data_processor.rb:4]: Can we rename this? Maybe something like `MINIMUM_GDPR_COMPLIANT_TOTAL`

Comment (tomas-stefano) [reply app/models/publications/monthly_statistics/statistics_data_processor.rb]: Done

File: app/models/publications/monthly_statistics/statistics_data_processor.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+module Publications
+  module MonthlyStatistics
+    class StatisticsDataProcessor
```
Comment (vassyz) [inline app/models/publications/monthly_statistics/statistics_data_processor.rb:3]: Shouldn't this have some specs? Seems the classes in this folder aren't unit tested.

Comment (tomas-stefano) [reply app/models/publications/monthly_statistics/statistics_data_processor.rb:3]: Done

**Review by JR-G — APPROVED** [review]

File: app/models/publications/monthly_statistics/statistics_data_processor.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        @status_data = status_data
+      end
+
+      def violates_gdpr?
+        @status_data[:this_cycle] < MINIMUM_GDPR_COMPLIANT_TOTAL ||
+          @status_data[:last_cycle] < MINIMUM_GDPR_COMPLIANT_TOTAL
+      end
+    end
+  end
+end
```
Comment (vassyz) [inline app/models/publications/monthly_statistics/statistics_data_processor.rb:17]: 🤩

**Review by vassyz — APPROVED** [review]

**Pull request merged**